### PR TITLE
Hera features

### PR DESCRIPTION
**Pull request by esnguyenvan**
This PR allows the user to pilot the x-position of HTP and VTP, which was previously hard coded within the geometry component (fixing issue  #35)

The major modification concerns the position of the HTP in case of T-tail:
- Previously, it is was set at 100% of the fuselage length
- Now it is set relatively to the VTP tip chord

Additionally, the unit test of high-lift effect was fixed. Previously, it would give false positive all the time. This changed quite significantly the test value, but when checking the new test values with the first commit (3f5996b90b15703612dc6b31d4684531ef8975a0) everything is in coherence.

### CODE REVIEW

**Review by ScottDelbecq — CHANGES_REQUESTED** [review]
Thanks for this PR!
I have a few comments and suggestions.
Regarding `wet_area_coeff` I did not find the source for checking the correct value.
I think this PR is the opportunity to include global positions for VTP and HTP like for the wing: https://github.com/fast-aircraft-design/FAST-OAD_CS25/blob/main/src/fastoad_cs25/models/geometry/geom_components/wing/wing_global_positions.py

**Review by christophe-david — CHANGES_REQUESTED** [review]
Thanks for the update.

Small updates are needed for the HL polar test and computation of HTP area.

For the tail positions, I made several comments about horizontal tail. They also apply to vertical tail.

File: src/fastoad_cs25/models/geometry/geom_components/ht/compute_horizontal_tail.py [hunk 1]
```diff
@@ -32,3 +42,9 @@ def setup(self):
         self.add_subsystem("ht_mac", ComputeHTMAC(), promotes=["*"])
         self.add_subsystem("ht_sweep", ComputeHTSweep(), promotes=["*"])
         self.add_subsystem("ht_cl_alpha", ComputeHTClalpha(), promotes=["*"])
+        self.add_subsystem("ht_local_positions", ComputeHTLocalPositions(), promotes=["*"])
+        self.add_subsystem(
+            "global_positions",
+            RegisterSubmodel.get_submodel(SERVICE_HORIZONTAL_TAIL_GEOMETRY_GLOBAL_POSITIONS),
+            promotes=["*"],
+        )
```
[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply src/fastoad_cs25/models/geometry/geom_components/ht/compute_horizontal_tail.py]: I agree... :)
Corrected in 1c2dbfe0a9cc8ec1a5f8ad6f6ecd6d82141a99de and 5a24b251c7a5d4a9407bf67e3ab642c9c177c40b .

File: src/fastoad_cs25/models/geometry/geom_components/ht/components/compute_ht_local_positions.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        taper_h = inputs["data:geometry:horizontal_tail:taper_ratio"]
+        x0_ht = inputs["data:geometry:horizontal_tail:MAC:at25percent:x:local"]
+        mac_ht = inputs["data:geometry:horizontal_tail:MAC:length"]
+        b_h = inputs["data:geometry:horizontal_tail:span"]
+        sweep_25_ht = inputs["data:geometry:horizontal_tail:sweep_25"]
+
+        x_mac = x0_ht - mac_ht * 0.25
+        y_mac = b_h / 2 / 3 * (1 + 2 * taper_h) / (1 + taper_h)
+        x_root = x_mac - y_mac * np.tan(sweep_25_ht / 180.0 * np.pi)
+        x_tip = x_root + b_h / 2 * np.tan(sweep_25_ht / 180.0 * np.pi)
```
[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply src/fastoad_cs25/models/geometry/geom_components/ht/components/compute_ht_local_positions.py]: Corrected in b7170e01e4dfb27b6c1173097dfb95808cc50184

File: src/fastoad_cs25/models/handling_qualities/tail_sizing/compute_ht_area.py [hunk 3]
```diff
@@ -107,11 +133,17 @@ def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
         ht_volume_coeff = cm_front_lg
 
         if tail_type == 1:
-            aero_centers_distance = fuselage_length - x_wing_aero_center
-            wet_area_coeff = 1.6
+            aero_centers_distance = (
+                vtp_aero_center
+                - vtp_x0
+                + vtp_span * np.tan(vtp_sweep_0)
```
[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply src/fastoad_cs25/models/handling_qualities/tail_sizing/compute_ht_area.py:139]: Corrected in 78f2a30fb430b23dccb0321e08fbd1a668a09f42

File: src/fastoad_cs25/models/geometry/geom_components/ht/components/compute_ht_local_positions.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    def compute(self, inputs, outputs):
+        taper_h = inputs["data:geometry:horizontal_tail:taper_ratio"]
+        x0_ht = inputs["data:geometry:horizontal_tail:MAC:at25percent:x:local"]
+        mac_ht = inputs["data:geometry:horizontal_tail:MAC:length"]
+        b_h = inputs["data:geometry:horizontal_tail:span"]
+        sweep_25_ht = inputs["data:geometry:horizontal_tail:sweep_25"]
+
+        x_mac = x0_ht - mac_ht * 0.25
+        y_mac = b_h / 2 / 3 * (1 + 2 * taper_h) / (1 + taper_h)
```
[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply src/fastoad_cs25/models/geometry/geom_components/ht/components/compute_ht_local_positions.py]: Corrected in 78f2a30fb430b23dccb0321e08fbd1a668a09f42

File: src/fastoad_cs25/models/geometry/geom_components/ht/components/compute_ht_local_positions.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        x_root = x_mac - y_mac * np.tan(sweep_25_ht / 180.0 * np.pi)
```
[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply src/fastoad_cs25/models/geometry/geom_components/ht/components/compute_ht_local_positions.py]: Corrected in [344d392](https://github.com/fast-aircraft-design/FAST-OAD_CS25/pull/36/commits/344d39271c4126069609a0b8c04490d9b442c468)

File: src/fastoad_cs25/models/handling_qualities/tail_sizing/compute_ht_area.py [hunk 6]
```diff
@@ -107,11 +133,17 @@ def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
...
+                + htp_leading_edge_position * vtp_tip_chord
+                + htp_x0
+            )
+            wet_area_coeff = 1.6  # TBD: these wet_area coefficients seem interverted
```
[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply src/fastoad_cs25/models/handling_qualities/tail_sizing/compute_ht_area.py]: Corrected in [c21f15a](https://github.com/fast-aircraft-design/FAST-OAD_CS25/pull/36/commits/c21f15a1953a1e2938eb5fe677b07c0c0b2e0c1c)

File: src/fastoad_cs25/models/geometry/geom_components/ht/components/compute_ht_local_positions.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        self.add_input(
+            "data:geometry:horizontal_tail:MAC:at25percent:x:local", val=np.nan, units="m"
+        )
+        self.add_input("data:geometry:horizontal_tail:MAC:length", val=np.nan, units="m")
+        self.add_input("data:geometry:horizontal_tail:span", val=np.nan, units="m")
+        self.add_input("data:geometry:horizontal_tail:sweep_25", val=np.nan, units="deg")
+
+        self.add_output("data:geometry:horizontal_tail:tip:leading_edge:x:local", units="m")
+        self.add_output("data:geometry:horizontal_tail:root:leading_edge:x:local", units="m")
+        self.add_output("data:geometry:horizontal_tail:MAC:leading_edge:x:local", units="m")
```
[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply src/fastoad_cs25/models/geometry/geom_components/ht/components/compute_ht_local_positions.py]: Corrected in a5e4a91

File: src/fastoad_cs25/models/handling_qualities/tail_sizing/compute_ht_area.py [hunk 8]
```diff
@@ -107,11 +133,17 @@ def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
...
         elif tail_type == 0:
             aero_centers_distance = htp_aero_center_ratio * fuselage_length - x_wing_aero_center
-            wet_area_coeff = 2.0
+            wet_area_coeff = 2.0  # TBD: these wet_area coefficients seem interverted
```
[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply src/fastoad_cs25/models/handling_qualities/tail_sizing/compute_ht_area.py]: I agree I did not change it but left a TODO in [c21f15a](https://github.com/fast-aircraft-design/FAST-OAD_CS25/pull/36/commits/c21f15a1953a1e2938eb5fe677b07c0c0b2e0c1c)

**Review by christophe-david — APPROVED** [review]
Ok for me. Thanks for the changes.

I suggest to wait for oncoming FAST-OAD-core v1.17.3 before merging. It will allow to check watchman tests on the branch.

Also, I still hesitate about keeping the variables `"data:geometry:vertical_tail:root:leading_edge:x:local"` and `"data:geometry:horizontal_tail:center:leading_edge:x:local"`, since their values are by definition 0. Yet it has the advantage of making clear the origin of the local reference frames, so it is maybe not so bad.

But if we decide to keep these variables, we should think about adding also `"data:geometry:wing:root:leading_edge:x:local"` for consistency.

**Pull request merged**